build.sbt

```
@@ -188,21 +201,24 @@ lazy val testsOutput = projectMatrix
     coverageEnabled := false
   )
   .defaultAxes(VirtualAxis.jvm)
-  .jvmPlatform(testTargetScalaVersions)
+  .jvmPlatform(buildScalaVersions :+ scala3)
   .disablePlugins(ScalafixPlugin)
 
 lazy val testkit = projectMatrix
   .in(file("scalafix-testkit"))
   .settings(
     moduleName := "scalafix-testkit",
     isFullCrossVersion,
-    libraryDependencies ++= Seq(
-      googleDiff,
-      scalatest
-    )
+    libraryDependencies += googleDiff,
+    libraryDependencies ++= {
+      if (!isScala3.value)
+        Seq(scalatest)
+      else
+        Seq("org.scalatest" %% "scalatest" % scalatest3V)
```

you can avoid repeating the artifact coordinates
```suggestion
        Seq("scalatest.withRevision(scalatest3V))
```
actually, since scalatest is used also in unit, we could probably have a `scalatestDep` in `ScalafixBuild`
```scala
    lazy val scalatestDeps = Def.setting {
      if (isScala3.value) Seq(scalatest.withRevision(scalatest3V))
      else Seq(scalatest)
    }
```

---

I have a doubt, why in unit we have scalatest with version 3.2.0:
```
    libraryDependencies ++= List(
      jgit,
      coursier,
      scalatest.withRevision(
        "3.2.0"
      ), // make sure testkit clients can use recent 3.x versions
      scalametaTeskit
    ),
```

while in `Dependencies.scala` we have a different version:
- `  val scalatestV = "3.0.8"`

In your `scalatestDeps` proposal we'd be using either version 3.0.8 (scalatestV) or 3.2.13 (scalatest3V)

---

That hardcoded "3.2.0" has no reason to be pinned, it can/should follow the latest version. So instead of `scalatestV` & `scalatest3V`, we could name `scalatestMinV` & `scalatestLatestV`.

It's OK that the scala3 artifacts are not compatible with Scalatest 3.0.x since there are new so there are no concerns of breaking changes. See https://github.com/scalacenter/scalafix/issues/1172#issuecomment-646621781 for the context. For the next breaking change version of scalafix, I believe we should remove support for Scalafix 3.0.x (it's been nearly 3 years since the old Scalatest 3.0.x API is deprecated) and just build against the latest, but let's not change that in that PR.